mat64/eigen.go

```diff
@@ -817,3 +820,49 @@ func (f EigenFactors) D() *Dense {
 	}
 	return dm
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (f EigenFactors) MarshalBinary() ([]byte, error) {
```

Comment:
It's easy for clients to extract the relevant data that might be reused later and serialise that.